src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -261,6 +261,16 @@ void IslandWindow::Initialize()
 
     _rootGrid = winrt::Windows::UI::Xaml::Controls::Grid();
     _source.Content(_rootGrid);
+
+    // initialize the taskbar object
+    HRESULT hr = CoCreateInstance(CLSID_TaskbarList, NULL, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(&_taskbar));
+    if (SUCCEEDED(hr))
+    {
+        if (SUCCEEDED(_taskbar->HrInit()))
+        {
+            _taskbarInitialized = true;
+        }
+    }
 }

```

Comment:
I'd prefer you to capture this in a _local_ and then `std::move` the local into `_taskbar`. That way, `_taskbar` is only set if it all succeeded. You can then also get rid of the boolean (because `_taskbar` will either be a fully-formed pointer, or it will be null).

Right now, if HrInit fails there's a hanging Taskbar object left lying around.